dev/archery/archery/crossbow/core.py

```diff
@@ -199,7 +199,7 @@ def credentials(self, url, username_from_url, allowed_types):
             raise CrossbowError(msg)
 
         if (allowed_types &
-                pygit2.credentials.GIT_CREDENTIAL_USERPASS_PLAINTEXT):
+                pygit2.credentials.CredentialType.USERPASS_PLAINTEXT):
```

Comment:
This is unrelated to the fix itself, but necessary to get `archery crossbow ...` working. pygit2 had an API change and we weren't testing against this.

Reply:
It seems that this requires pygit2 1.14.0 or later. Could you also update required pygit2?

```diff
diff --git a/dev/archery/setup.py b/dev/archery/setup.py
index 2ecc72e04e..f502876856 100755
--- a/dev/archery/setup.py
+++ b/dev/archery/setup.py
@@ -29,7 +29,7 @@ jinja_req = 'jinja2>=2.11'
 
 extras = {
     'benchmark': ['pandas'],
-    'crossbow': ['github3.py', jinja_req, 'pygit2>=1.6.0', 'requests',
+    'crossbow': ['github3.py', jinja_req, 'pygit2>=1.14.0', 'requests',
                  'ruamel.yaml', 'setuptools_scm<8.0.0'],
     'crossbow-upload': ['github3.py', jinja_req, 'ruamel.yaml',
                         'setuptools_scm'],
```

(This may break CI. So I think that separated PR is better.)

Reply:
Sure, happy to move it to a separate PR

Reply:
Done: https://github.com/apache/arrow/pull/40269